### PR TITLE
Rene/issue 176 lints (package names and folder structure)

### PR DESCRIPTION
**Pull request by Leopard2A5**
Implement lint that makes sure the package hierarchy is mirrored by the folder structure.

Screenshot 2024-08-08 at 14 56 12

### CODE REVIEW

File: src/validation/rules/package_hierarchy.rs [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+mod tests {
+    use super::*;
+
+    static EXPECTED_HELP: &str = "Package names should be mirrored in folder structure, eg mypackage.subpackage should live in proto/subpackage.";
+
+    #[test]
+    fn file_name_should_not_matter() {
+        let package = Package {
+            name: "root".into(),
+            files: vec!["proto/file_not_called_root.proto".into()],
```
Comment (mara-schulke) [inline src/validation/rules/package_hierarchy.rs:78]: This is actually fine!

Comment (mara-schulke) [reply src/validation/rules/package_hierarchy.rs:78]: Ie. only subfolders in the proto folder denote subpackages

proto/*.proto -> pkg x;

proto/y/*.proto -> pkg x.y;

File: src/validation/rules/package_hierarchy.rs [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+                "proto/not_subpackage/file.proto".into(),
+                "proto/foo/bar/file.proto".into(),
+            ],
+            entities: Default::default(),
+        };
+        assert_eq!(
+            PackageHierarchy.check_package(&package),
+            &[
+                Violation {
+                rule: "PackageHierarchy".into(),
```
Comment (mara-schulke) [inline src/validation/rules/package_hierarchy.rs:147]: formatting?

File: src/validation/rules/lib_package.rs [hunk 3]
```diff
@@ -8,10 +8,6 @@ impl Rule for LibPackage {
         "Make sure that lib packages don't contain service definitions."
     }
 
-    fn rule_level(&self) -> Level {
```
Comment (mara-schulke) [inline src/validation/rules/lib_package.rs:11]: What's the implication / reason behind this change?

**Review by mara-schulke — APPROVED** [review]

File: src/validation/rules/package_hierarchy.rs (hunk 2 above)
Comment (Leopard2A5) [reply src/validation/rules/package_hierarchy.rs]: done :)

File: src/validation/rules/lib_package.rs (hunk 3 above)
Comment (Leopard2A5) [reply src/validation/rules/lib_package.rs:11]: That the default impl of `Rule` already returns `Level::Error`, so going with the default and not writing code that has no effect beyond the default.

**Pull request merged**